sdk/lib/rtl/assert.c

```diff
@@ -35,12 +35,28 @@ RtlAssert(IN PVOID FailedAssertion,
     for (;;)
     {
         /* Print the assertion */
-        DbgPrint("\n*** Assertion failed: %s%s\n"
-                 "***   Source File: %s, line %ld\n\n",
-                 Message != NULL ? Message : "",
-                 (PSTR)FailedAssertion,
-                 (PSTR)FileName,
-                 LineNumber);
+        if (Message == NULL)
+        {
+            DbgPrint("\n"
+                      "*** Assertion failed: %s\n"
+                      "***   Source File: %s, line %lu\n"
+                      "\n",
+                     (PSTR)FailedAssertion,
+                     (PSTR)FileName,
+                     LineNumber);
+        }
+        else
+        {
+            DbgPrint("\n"
+                      "*** Assertion failed: %s\n"
+                      "***  Message: %s\n"
```

Comment:
This is a public API with user-visible behavior. You can't change this (unless you have a test that shows Windows does the same... which it doesn't).

Reply:
All right. Which part(s) cannot be changed? How does Windows display such assertions?
ReactOS current display as `MessageAssertion` looks rather odd, even when `Message` ends with an `\n`...

Reply:
It displays **exactly** as it was in the original code before your changes.

Reply:
So basically, keeping the "\n" as this was in the original files helped to put the "expression" in a separate line; an alternative would be to put a space, or whatever other separator. But **not** removing the "\n" !

Reply:
Thanks for the confirmation! I will improve these the other way round then.